library/common/engine.cc

```diff
@@ -94,14 +95,10 @@ Engine::~Engine() {
 }
 
 void Engine::recordCounter(std::string elements, uint64_t count) {
-  if (server_) {
+  if (server_ && external_scope_) {
     server_->dispatcher().post([this, elements, count]() -> void {
-      static const std::string client = "client";
-      absl::string_view prefix{client};
       absl::string_view dynamic_elements{elements};
```

Comment:
nit: can we check if the string view can be built in place in the call below?

Reply:
I checked it; it doesn't work. In fact, there's another transparent type being created here, which is an `const InlinedVector<variant<Envoy::Stats::StatName, Envoy::Stats::DynamicName>`.